src/xmltv.js

```diff
@@ -1,12 +1,15 @@
 const XMLWriter = require('xml-writer')
 const fs = require('fs')
+const CacheImageService = require('./cache-image-service');
+let dataBase = undefined;
 
 module.exports = { WriteXMLTV: WriteXMLTV, shutdown: shutdown }
 
 let isShutdown = false;
 let isWorking = false;
 
-async function WriteXMLTV(json, xmlSettings, throttle ) {
+async function WriteXMLTV(json, xmlSettings, throttle, db) {
```

Comment:
Instantiate the CacheImageService outside all of this. And pass it to writeXMLTV function instead of passing db.